src/types.ts

```diff
@@ -48,6 +47,11 @@ export interface ProviderConfig {
    */
   user?: LDUser;
 
+  /**
+   * If set to true, the ldClient will not be initialzed until the user prop has been defined.
+   */
+  waitForUserToInitializeClient?: boolean;
+
```

Comment:
What do you think of calling this new prop `deferInitialization` instead? Reason being, there could be other reasons for doing this in the future in addition to waiting for the user object.

Reply:
Yes, I'm happy to change it. Though, would you want a different flag to distinguish what we are deferring initialization for? For example, in the case that we wanted to defer initialization until `props.foo` is provided even if `props.user` has been loaded.

Reply:
At this time, I think we can safely assume deferring initialization only refers to the user and use the documentation to support this.